### PR TITLE
feat(tsconfig): add support for ts 4.6 and ng14

### PR DESCRIPTION
**Pull request by mhenkens**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x The commit message follows our guidelines: https://github.com/NationalBankBelgium/code-style/blob/master/CONTRIBUTING.md#-commit-message-guidelines
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Add support for TS 4.6 and angular 14

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: lib/eslint/8.30.x/.eslintrc.json
Comment (SuperITMan) [inline lib/eslint/8.30.x/.eslintrc.json:1]: Could you remove this file ?

File: lib/eslint/8.30.x/README.md
Comment (SuperITMan) [inline lib/eslint/8.30.x/README.md:1]: Could you remove this file ?

File: lib/eslint/8.30.x/package.json
Comment (SuperITMan) [inline lib/eslint/8.30.x/package.json:1]: Could you remove this file ?

File: lib/tsconfig/4.6.x/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    "strictFunctionTypes": true,
+    "strictPropertyInitialization": true,
+    "strictNullChecks": true,
+    "stripInternal": true,
+    "skipLibCheck": true,
+    "suppressImplicitAnyIndexErrors": true,
+    "suppressExcessPropertyErrors": false,
+    "useDefineForClassFields": true,
+    "useUnknownInCatchVariables": true,
+    "target": "es2020",
```
Comment (SuperITMan) [inline lib/tsconfig/4.6.x/tsconfig.json:54]: Could you add a comment about this change (previously it was es2015) in your commit message ?

Does the Angular recommends/forces to use the targert `es2020` ?

**Review by SuperITMan — CHANGES_REQUESTED** [review]
Some remarks. 

Could you the edit the tsconfig/README.md file in order to add the new versions and to adapt the latest version ?

**Review by ageorges-nbb — APPROVED** [review]

**Review by SuperITMan — APPROVED** [review]

**Pull request merged**